frontend/ui/widget/imageviewer.lua

```diff
@@ -271,12 +271,9 @@ function ImageViewer:init()
         end
     end
 
-    if self._images_list then
+    if self._images_list and self._images_list_nb > 1 then
```

Comment:
Was this really an issue ? It defaults to `_images_list = nil`. Who/where/what could provide a `_images_list = {}` and the need to check for self._images_list_nb ?
Oh, ok, it's just for when there's a single image. So, Who/where/what could provide a single item _images_list ?

Also please check you get that progress bar with an animated gif, ie. https://upload.wikimedia.org/wikipedia/commons/c/c2/HRR.gif

Reply:
> Who/where/what could provide a single item _images_list ?

Yesterday OPDS book cover, via `OPDSPSE:streamPages`.

Reply:
We now got a crash when viewing stream pages in OPDSPSE because of their hack:
https://github.com/koreader/koreader/blob/2b18a7e90ffbd6c1e8733306620d5622336e7b04/plugins/opds.koplugin/opdspse.lua#L165-L166
`_image_list` passed to ImageViewer contains one image only, but number of images is changed to the total number of pages in a book. `setmetatable` is used to get the next member of the array (download the next image).
So in `ImageViewer:new` we do not create the progress bar (because number of images is 1), but then `ImageViewer:update` tries to update it (because number of images == count).

Reply:
As you feel.
Or you could add `ImageViewer.always_show_progressbar = true` (and have it used by opdspse) to bypass your changes, or something like that.